lightning/src/ln/channelmonitor.rs

```diff
@@ -422,6 +422,7 @@ pub(crate) enum InputMaterial {
 	},
 	Funding {
 		channel_value: u64,
+		funding_redeemscript: Script,
 	}
```

Comment:
2c09663

We do store local_channel_pubkeys+remote_channel_pubkeys in `InMemoryChannelKeys`, can't you derive script there instead ?

Reply:
See-also #606.

Reply:
Hmm, we don't expose the remote_channel_pubkeys in ChannelKeys, nor do I really think we should. We could maybe move to caching that in the ChannelMonitor and passing it in here, but for now might as well just leave it as caching the script.

Reply:
I pretty agree with other lines in #606, for now I think we can cache the script. I expect signer to verify remote keys, so why caching in ChannelMonitor?

Reply:
Ehh, I was just thinking try to be less prescriptive on the signer, who knows the signer may be purely remote/hardware/etc, so no need to be too strict about what it provides.